Neos.Flow/Classes/Configuration/Loader/RoutesLoader.php

```diff
@@ -84,20 +84,28 @@ public function load(array $packages, ApplicationContext $context): array
     protected function includeSubRoutesFromSettings(array $routeDefinitions, array $routeSettings): array
     {
         $sortedRouteSettings = (new PositionalArraySorter($routeSettings))->toArray();
-        foreach ($sortedRouteSettings as $packageKey => $routeFromSettings) {
+        foreach ($sortedRouteSettings as $configurationKey => $routeFromSettings) {
             if ($routeFromSettings === false) {
                 continue;
             }
-            $subRoutesName = $packageKey . 'SubRoutes';
-            $subRoutesConfiguration = ['package' => $packageKey];
+            if (isset($routeFromSettings['providerFactory'])) {
+                $routeDefinitions[] = [
+                    'name' => $configurationKey,
+                    'providerFactory' => $routeFromSettings['providerFactory'],
+                    'providerOptions' => $routeFromSettings['providerOptions'] ?? [],
+                ];
+                continue;
```

Comment:
i was trying to write some functional test for this but doesnt this `continue` here prohibit that `Neos.Flow` provides _real_ routes.yaml AND routes from annotation?

Currently we use

```yaml
Neos:
  Flow:
    mvc:
      routes:
        'Neos.Flow': TRUE
```

but when changing this to

```yaml
Neos:
  Flow:
    mvc:
      routes:
        'Neos.Flow':
          providerFactory: ...
```

this will not include the defined routes of _routes.yaml_ no?

Or is the idea to use a "virtual" package / separate key like

```yaml
Neos:
  Flow:
    mvc:
      routes:
        'Neos.Flow:FromAnnotation':
          providerFactory: ...
```

Reply:
I thought the idea is the latter

Reply:
Yeah though so too and also the code looks like it. Will document this as recommended.

Reply:
If you want both (which is unlikely for 3rd party packages) you'll write:
```yaml
Neos:
  Flow:
    mvc:
      routes:
        'Neos.Flow': true
        'Neos.Flow.attributes':
          providerFactory: ...
```

Reply:
But a 3rd party package would not use the Neos.Flow namespace ofc and instead for example:
```yaml
Neos:
  Flow:
    mvc:
      routes:
        'Some.Package':
          before: 'Neos.Flow'
          providerFactory: ...
```